src/v/kafka/server/group_manager.cc

```diff
@@ -682,6 +682,84 @@ ss::future<> group_manager::reload_groups() {
     co_await ss::when_all_succeed(futures.begin(), futures.end());
 }
 
+ss::future<result<model::offset>> group_manager::set_blocked_for_groups(
+  const model::ntp& co_ntp,
+  const chunked_vector<kafka::group_id>& group_ids,
+  bool blocked) {
+    if (!_feature_table.local().is_active(
+          features::feature::consumer_groups_migrations)) {
+        vlog(
+          cg_klog.warn,
+          "set blocked request for {} failed - consumer groups migrations "
+          "feature is not active",
+          co_ntp);
+        co_return cluster::errc::feature_disabled;
+    }
+    auto p = get_attached_partition(co_ntp);
+    if (!p) {
+        vlog(
+          cg_klog.warn,
+          "set blocked request for {} failed - partition not found",
+          co_ntp);
+        co_return cluster::errc::partition_not_exists;
+    }
+    if (!p->catchup_lock->try_read_lock()) {
+        vlog(
+          cluster::txlog.trace,
+          "can't set blocked: coordinator_load_in_progress");
+        co_return cluster::tx::errc::coordinator_load_in_progress;
+    }
+    p->catchup_lock->read_unlock();
+    auto holder = p->catchup_lock->hold_read_lock();
+
+    storage::record_batch_builder builder(
+      model::record_batch_type::group_block, model::offset{0});
+    for (const auto& group_id : group_ids) {
+        group_block{group_id, blocked}.add_to_batch_builder(builder);
+    }
+
+    auto result = co_await p->partition->raft()->replicate(
+      p->term,
+      std::move(builder).build(),
+      raft::replicate_options(raft::consistency_level::quorum_ack));
+
+    if (!result) {
+        co_return result.error();
+    }
+
+    if (blocked) {
+        auto last_error = cluster::tx::errc::none;
+        co_await ss::max_concurrent_for_each(
+          group_ids, 5, [this, &last_error](auto& group_id) {
+              auto it = _groups.find(group_id);
+              if (it == _groups.end()) {
+                  return ss::make_ready_future<>();
+              }
+              return it->second->abort_txes(true).then(
```

Comment:
hmmm.. what if block replication succeeds and abort fails? Wondering if we should reverse these operations.. (perhaps cache term before doing it to avoid ABA leadership problems)

wait until all open transactions are aborted, ensure no open transaction (has_transactions_in_progress()), then replicate block.